server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -897,12 +897,47 @@ public Owner updateOwner(@PathParam("owner_key") @Verify(Owner.class) String key
             }
         }
 
-        // Update the autobindDisabled field if the incoming value is null.
+        // Update the autobindDisabled field if the incoming value is not null.
         if (owner.getAutobindDisabled() != null) {
             toUpdate.setAutobindDisabled(owner.getAutobindDisabled());
         }
 
+        // Update the contentAccess field if the incoming value is not null.
+        if (owner.getContentAccessMode() != null) {
+            StringTokenizer st = new StringTokenizer(toUpdate.getContentAccessModeList(), ",");
+            boolean allowed = false;
+            while (st.hasMoreElements()) {
+                if (st.nextElement().equals(owner.getContentAccessMode()) ||
+                    "".equals(owner.getContentAccessMode())) {
+                    allowed = true;
+                    break;
+                }
+            }
+            if (allowed) {
+                String before = toUpdate.getContentAccessMode();
+                String after = owner.getContentAccessMode();
+                toUpdate.setContentAccessMode(owner.getContentAccessMode());
+                if (!after.equals(before)) {
+                    toUpdate.setContentAccessModeDirty(true);
+                }
+            }
+            else {
+                throw new BadRequestException(
+                    i18n.tr("The content access mode is not allowed for this owner."));
+            }
+        }
+
         ownerCurator.merge(toUpdate);
+
+        if (toUpdate.isContentAccessModeDirty()) {
+            try {
+                ownerManager.refreshOwnerForContentAccess(toUpdate);
```

Comment:
How long does this take for an owner with large numbers of subscriptions & large numbers of consumers?